code/game/machinery/quantum_pad.dm

```diff
@@ -0,0 +1,148 @@
+/obj/machinery/quantumpad
+	name = "quantum pad"
+	desc = "A bluespace quantum-linked telepad used for teleporting objects to other quantum pads."
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "qpad-idle"
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 200
+	active_power_usage = 5000
+	var/teleport_cooldown = 400 //30 seconds base due to base parts
+	var/teleport_speed = 50
+	var/last_teleport //to handle the cooldown
+	var/teleporting = 0 //if it's in the process of teleporting
+	var/power_efficiency = 1
+	var/obj/machinery/quantumpad/linked_pad = null
+
+/obj/machinery/quantumpad/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/quantumpad(null)
+	component_parts += new /obj/item/weapon/ore/bluespace_crystal/artificial(null)
+	component_parts += new /obj/item/weapon/stock_parts/capacitor(null)
+	component_parts += new /obj/item/weapon/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stack/cable_coil(null, 1)
+	RefreshParts()
+
+/obj/machinery/quantumpad/Destroy()
```

Comment:
This is technically not enough for references to clear. No, I don't expect you to fix it, because you *technically* need a list of multi-tools and quantum pads to properly clear all references on this--and I don't expect you to do that in this PR; just making a note of it for anyone who *does* care enough.